components/email-mgt/org.wso2.carbon.email.mgt/src/main/java/org/wso2/carbon/email/mgt/config/EmailTemplateManagerImpl.java

```diff
@@ -0,0 +1,358 @@
+
+        // validate the email template object before processing it.
+        validateEmailTemplate(emailTemplate, tenantDomain);
+
+        Resource templateResource = I18nEmailUtil.createTemplateResource(emailTemplate);
+        String templateTypeDisplayName = emailTemplate.getTemplateDisplayName();
+        String templateType = I18nEmailUtil.getNormalizedName(templateTypeDisplayName);
+        String locale = emailTemplate.getLocale();
+
+        String path = TEMPLATE_BASE_PATH + PATH_SEPARATOR + templateType; // template type root directory
+        try {
+            // check whether a template type root directory exists
+            if (!resourceMgtService.isResourceExists(path, tenantDomain)) {
+                // we add new template type with relevant properties
+                addEmailTemplateType(templateTypeDisplayName, tenantDomain);
+                if (log.isDebugEnabled()) {
+                    String msg = "Creating template type %s in %s tenant registry.";
+                    log.debug(String.format(msg, templateTypeDisplayName, tenantDomain));
+                }
+            }
+
+            resourceMgtService.putIdentityResource(templateResource, path, tenantDomain, locale);
+        } catch (IdentityRuntimeException ex) {
+            String errorMsg = "Error when adding new email template of %s type, %s locale to %s tenant registry.";
+            handleServerException(String.format(errorMsg, templateResource, locale, tenantDomain), ex);
+        }
+    }
+
+
+    @Override
+    public void deleteEmailTemplate(String templateTypeName, String localeCode, String tenantDomain) throws
+            I18nEmailMgtException {
+        // validate the name and locale code.
+        if (StringUtils.isBlank(templateTypeName)) {
+            throw new I18nEmailMgtClientException("Cannot Delete template. Email displayName cannot be null.");
+        }
+
+        if (StringUtils.isBlank(localeCode)) {
+            throw new I18nEmailMgtClientException("Cannot Delete template. Email locale cannot be null.");
+        }
+
+        String templateType = I18nEmailUtil.getNormalizedName(templateTypeName);
+        String path = TEMPLATE_BASE_PATH + PATH_SEPARATOR + templateType;
+
+        try {
+            resourceMgtService.deleteIdentityResource(path, tenantDomain, localeCode);
+        } catch (IdentityRuntimeException ex) {
+            String msg = String.format("Error deleting %s:%s template from %s tenant registry.", templateTypeName,
+                    localeCode, tenantDomain);
+            handleServerException(msg, ex);
+        }
+    }
+
+
+    @Override
+    public void addDefaultEmailTemplates(String tenantDomain) throws I18nEmailMgtException {
+        // before loading templates we check whether they already exist.
+        try {
+            if (resourceMgtService.isResourceExists(TEMPLATE_BASE_PATH, tenantDomain)) {
+                if (log.isDebugEnabled()) {
```

Comment:
Can refactor the code with !(resourceMgtService.isResourceExists(TEMPLATE_BASE_PATH, tenantDomain))